webknossos-datastore/app/com/scalableminds/webknossos/datastore/explore/NgffExplorationUtils.scala

```diff
@@ -259,6 +261,42 @@ trait NgffExplorationUtils extends FoxImplicits {
       layerAndVoxelSizeTuples = layers.map((_, VoxelSize(voxelSizeFactor, unifiedAxisUnit)))
     } yield layerAndVoxelSizeTuples
 
+  protected def getTranslation(multiscale: NgffMultiscalesItem): Option[List[CoordinateTransformation]] = {
+    val is2d = !multiscale.axes.exists(_.name == "z")
+    val baseTranslation = if (is2d) List(1.0, 1.0) else List(1.0, 1.0, 1.0)
+    if (!multiscale.datasets.head.coordinateTransformations.exists(_.`type` == "translation")) {
```

Comment:
This will throw exceptions if multiscale.datasets is empty. Could we use headOption, with the whole function returning None if that is the case?

Reply:
Ok, is there a reasonable case where datasets would be empty?

Reply:
Not really, I’d just prefer the scala-esque handling of the empty case over exceptions.